t/lib/WTSI/NPG/HTS/PacBio/Sequel/RunDeleteMonitorTest.pm

```diff
@@ -0,0 +1,169 @@
+     $wh_db_file);
+}
+sub teardown_databases : Test(shutdown) {
+  $wh_schema->storage->disconnect;
+}
+
+sub setup_test : Test(setup) {
+  my $irods = WTSI::NPG::iRODS->new(environment          => \%ENV,
+                                    strict_baton_version => 0);
+
+  $irods_tmp_coll =
+    $irods->add_collection("RunPublisherTest.$pid.$test_counter");
+  $test_counter++;
+}
+
+sub teardown_test : Test(teardown) {
+  my $irods = WTSI::NPG::iRODS->new(environment          => \%ENV,
+                                    strict_baton_version => 0);
+  $irods->remove_collection($irods_tmp_coll);
+}
+
+
+sub require : Test(1) {
+  require_ok('WTSI::NPG::HTS::PacBio::Sequel::RunDeleteMonitor');
+}
+
+sub delete_runs : Test(7) {
+  my $uri    = URI->new($server->uri . 'QueryJobs');
+  my $client = WTSI::NPG::HTS::PacBio::Sequel::APIClient->new
+     (default_interval => 10000,);
+  $client->{'runs_api_uri'} = $uri;
+
+  my $irods = WTSI::NPG::iRODS->new(environment          => \%ENV,
+                                    strict_baton_version => 0);
+  my $drirods = WTSI::NPG::DriRODS->new(environment          => \%ENV,
+                                        strict_baton_version => 0);
+  my $dest_coll = $irods_tmp_coll;
+
+  ## create tmp runfolder which will be deleted later
+  my $run_name   = 'r54097_20170727_165601';
+  my $well       = '1_A02';
+  my $data_path  = catdir('t/data/pacbio/sequel', $run_name, $well);
+
+  my $runfolder_path = catdir($tmp_dir,$run_name);
+  mkdir $runfolder_path;
+  my $runfolder_data = catdir($runfolder_path,$well);
+  dircopy($data_path,$runfolder_path) or die $!;
+
+  ## publish data
+  my $monitor = WTSI::NPG::HTS::PacBio::Sequel::RunMonitor->new
+    (api_client         => $client,
+     dest_collection    => $dest_coll,
+     irods              => $irods,
+     local_staging_area => $tmp_dir,
+     mlwh_schema        => $wh_schema);
+
+  my ($num_jobs, $num_processed, $num_errors) =
+    $monitor->publish_completed_runs;
+
+  cmp_ok($num_jobs, '==', scalar @{$test_response},
+         'Correct number of runs to publish');
+  cmp_ok($num_processed, '==', $num_jobs, 'All runs processed');
+  cmp_ok($num_errors, '==', 0, 'No error in any run published');
+
+  ## delete run
+  my $deletable = WTSI::NPG::HTS::PacBio::Sequel::RunDeleteMonitor->new
+    (api_client         => $client,
+     check_format       => 0,
+     dest_collection    => $dest_coll,
+     irods              => $drirods,
+     local_staging_area => $tmp_dir,
+     mlwh_schema        => $wh_schema);
+
+  my ($dnum_runs, $dnum_processed, $dnum_deleted, $dnum_errors) = 
+      $deletable->delete_runs();
+
+  cmp_ok($dnum_runs, '==', scalar @{$test_response},
+         'Correct number of runs to delete');
+  cmp_ok($dnum_processed, '==', $num_jobs, 'All run folders processed');
+  cmp_ok($dnum_processed, '==', $num_jobs, 'All run folders deleted');
+  cmp_ok($dnum_errors, '==', 0, 'No error in any run deleted');
+
```

Comment:
This could use another test to show that it doesn't delete a run that is not fully uploaded. e.g. the test harness could delete one file from the published run in iRODS, before the check.

Reply:
Good point - Thanks.